File: Part 2 - Regression/Section 8 - Decision Tree Regression/Decision_Tree.py
# -*- coding: utf-8 -*-
"""
Created on Sun Jul 15 15:11:49 2018

@author: Vinayak_S
"""

#importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

#Reading data
df = pd.read_csv('Position_Salaries.csv')
X = df.iloc[:,1:-1].values
y = df.iloc[:,-1].values

#Implementing Decision Tree Regressor
from sklearn.tree import DecisionTreeRegressor
regressor = DecisionTreeRegressor(random_state = 0)
regressor.fit(X,y)

#Predicting using Decision Tree Regressor
y_pred = regressor.predict(6.5)

#Visualisation of the Model
X_grid = np.arange(min(X), max(X), 0.01).reshape(-1,1)
plt.scatter(X,y)
plt.plot(X_grid,regressor.predict(X_grid))
plt.title('Level vs Salary(Decision Tree Regression)')
plt.xlabel('Level')
plt.ylabel('Salary')

File: Part 2 - Regression/Section 8 - Decision Tree Regression/Decision_Tree.R
#Reading the data
df = read.csv('Position_Salaries.csv')
df = df[2:3]

#Fitting Decision Tree Regressor to the data
#install.packages('rpart')
library(rpart)
regressor = rpart(formula = Salary ~ .,
                  data = df,
                  control = rpart.control(minsplit = 1))

#Predicting using Decision Tree
test = data.frame(Level = 6.5)
y_pred = predict(regressor, newdata = test)

#Visualising the Decision Tree Regression
library(ggplot2)
X_grid = seq(min(df$Level), max(df$Level), 0.01)
ggplot()+
  geom_point(aes(x = df$Level, y = df$Salary),
             colour = 'red')+
  geom_line(aes(x = X_grid, y = predict(regressor, 
                                        newdata = data.frame(Level = X_grid))),
            colour = 'blue')+
  ggtitle('Level vs Salary (Decision Tree Regression Model)')+
  xlab('Level')+
  ylab('Salary')
